device-simulator: integrate with current iotivity-lite

Danielius1922 opened the pull request:
- clone and compile current iotivity-lite repository
- fix tests after changes in iotivity-lite

Danielius1922 committed:
device-simulator: integrate with current iotivity-lite

- clone and compile current iotivity-lite repository
- fix tests after changes in iotivity-lite

ondrejtomcik commented on device-simulator/Dockerfile:
```diff
@@ -1,14 +1,21 @@
+ARG MBEDTLS_COMMIT=d81c11b8ab61fd5b2da8133aa73c5fe33a0633eb
+ARG TINYCBOR_COMMIT=70aba6ba51881e5b8d108c105a17ed9cdee6bc30
+RUN git clone --recursive https://github.com/iotivity/iotivity-lite.git && \
+  cd /iotivity-lite && git checkout $IOTIVITY_COMMIT && \
```
Why do you specify custom submodule commits?

Danielius1922 committed:
grpc-gateway: add resource test cases

Add test cases for dynamically creating, deleting, retrieving
and updating resources in a collection.

jkralik commented on pkg/ocf/ocf.go:
```diff
@@ -0,0 +1,23 @@
+	OC_IF_RW       = "oic.if.rw"     // read-write
+	OC_IF_A        = "oic.if.a"      // actuator
+	OC_IF_S        = "oic.if.s"      // sensor
+	OC_IF_CREATE   = "oic.if.create" // create
```
I think we can move these constants to https://github.com/plgd-dev/device/tree/main/schema as an interfaces package

jkralik commented on pkg/ocf/ocf.go:
```diff
@@ -0,0 +1,23 @@
+	OC_RT_P   = "oic.wk.p"   // platform
+	OC_RT_D   = "oic.wk.d"   // device
+	OC_RT_CON = "oic.wk.con" // configuration
+	OC_RT_COL = "oic.wk.col" // collection
```
I think there are defined in `https://github.com/plgd-dev/device/tree/main/schema`. If it is not then we can define them.

jkralik commented on pkg/ocf/ocf.go:
```diff
@@ -0,0 +1,23 @@
+	OC_RT_CON = "oic.wk.con" // configuration
+	OC_RT_COL = "oic.wk.col" // collection
+
+	OC_RT_DEVICE_CLOUD = "oic.d.cloudDevice"
```
this test resource type so move it to `hub/test/config`

jkralik commented on pkg/ocf/ocf.go:
```diff
@@ -0,0 +1,23 @@
+
+	OC_RT_DEVICE_CLOUD = "oic.d.cloudDevice"
+
+	OC_RT_RESOURCE_SWITCH = "oic.r.switch.binary"
```
this could be moved to https://github.com/plgd-dev/device/tree/main/schema or  `hub/test/config`

Danielius1922 committed:
Integrate constants from plgd-dev/device

Use constants for resource types and resources URIs
from plgd-dev/device packages.

Danielius1922 committed:
Remove device-simulator

Use dockerfile from iotivity-lite repository instead
of local container.
